### ISSUE TITLE
Notepad++ constantly downgrades itself after a while for no reason

### ISSUE THREAD

**Issue by Andre601** [notepad-plus-plus/notepad-plus-plus#13180]
### Description of the Issue

For some unknown reason is Notepad++ constantly being downgraded to an older version (concidentally one that doesn't have the dark-mode built in) and I cannot figure out for the life of my why this happens exactly.

I at most could blame my anti-virus for it as it has a feature to check for updates of programms and start an installation of them in the background. However, it so far always installed the proper, latest release of the programm and this issue never appeared to me in the past.

### Steps to Reproduce the Issue

1. Update Notepad++ to whatever the latest version is
2. Use it for a few days
3. Notice how it one day is downgraded

### Expected Behavior

Notepad++ should stay on the latest version.

### Actual Behavior

Notepad++ is constantly being downgraded after a few days, requiring me to update it again and again.

### Debug Information

Notepad++ v8.4.8   (32-bit)
Build time : Dec 24 2022 - 19:30:12
Path : C:\Program Files (x86)\Notepad++\notepad++.exe
Command Line : "C:\Program Files (x86)\Notepad++\change.log" 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.2604
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)

Something I would like to point out here is the fact that Notepad++ is, according to the debug above, a 32-bit installation, while the Settings page listing the programms is listing it as "Notepad++ (64-Bit x64)", which confuses me a lot... Is it a 64-Bit or a 32-Bit installation now?

![grafik](https://user-images.githubusercontent.com/11576465/220109379-530a3d4f-4b2d-4662-bb9b-8f9f12d7b541.png)

Additionally is latest version on the site 8.4.9, while latest installation through the "Update Notepad++" prompt seems to be 8.4.8?
Are installations broken or did you just not bump the version properly, causing this entire issue?

**Comment by Andre601** [comment]
Okay, I may have found a possible issue... For some reason was NP++ installed twice on my PC: Once as 64 installation and once as 32... Tho, this imo still doesn't solve the above issue given that the 32-Bit installation has been updated to latest available but is 8.4.8 still...

Will do a re-install of the software and see if that helps...

**Comment by alankilborn** [comment]
>Is it a 64-Bit or a 32-Bit installation now?

It is whichever you want; see the choice on the download page for the software, [HERE](https://notepad-plus-plus.org/downloads/v8.4.9/) for v.8.4.9 (for example).

------------------------

>Additionally is latest version on the site 8.4.9, while latest installation through the "Update Notepad++" prompt seems to be 8.4.8?

This is FAQ.  As of this writing, 8.4.9 was "recently" released.  There's a "cooling off period" before updating would update to 8.4.9 rather than 8.4.8 -- and that period hasn't expired yet.

------------------------

>Are installations broken or did you just not bump the version properly, causing this entire issue?

Nothing is broken.

**Comment by Andre601** [comment]
Yeah, it probably was the fact that somehow 2 versions were installed and that they both had caused conflicts between each other... Idk.
I'll close this for now, but re-open should the issue appear again.

**Comment by Andre601** [comment]
Small Update. It happened again. Notepad++ was downgraded again without any real explanation why. This time however are there not 2 installations present.

One thing I noticed is, that when installing the update does NP++ default to `C:\...` which isn't the default drive I have it installed on... Is there any way I can tell it to use the other place for installation when the next update would be provided?

Also, is there any way to figure out why it downgraded itself (or what downgraded it)?

**Comment by alankilborn** [comment]
Confession:  I've never installed Notepad++.  I've only used portable versions.  IMO there is zero reason to install it...but that's just me.  Given that, I reserve the right to bail out on this conversation at any time.  :-)  Or to be told I don't know what I'm talking about (because I don't).  But I think I can ask some questions that might help...

So 2 days ago you had a stable situation (it seems, or you wouldn't have closed this).  Then something happened to change that status.  Did you get an update notification, and you agreed to it?  Or was this a totally silent downgrade (in which case I'd ask how you even knew it happened)?

So now you're back to running the downgraded version?  Something _before_ 8.4.8?  If so, while that's running, can you post the _Debug Information_ from that version?

>One thing I noticed is, that when installing the update does NP++ default to C:\... which isn't the default drive I have it installed on... Is there any way I can tell it to use the other place for installation when the next update would be provided?

I think there was a recent change to the installer so an existing location would be picked up and ran with...but maybe that change was too recent to be in effect for your older downgrade installer when that runs?  (Man, this is getting confusing).

**Comment by Andre601** [comment]
> So 2 days ago you had a stable situation (it seems, or you wouldn't have closed this). Then something happened to change that status. Did you get an update notification, and you agreed to it? Or was this a totally silent downgrade (in which case I'd ask how you even knew it happened)?

I wouldn't call it "stable". I just noticed that I had two installations on my PC (A 32-Bit and a 64-Bit version), which I both deinstalled and then installed the 64-Bit version. I closed the issue because I thought this could've been the issue (A mixum of versions where Notepad++ got confused which version it now was (or where it checks)).
But the issue didn't fix itself there.

It also was a silent downgrade. I opened a file in Notepad++ and was greeted with a bright white, which shouldn't happen as I used the built-in dark theme of the latest version. A Quick check using the settings showed that the dark mode was gone, which meant it somehow got downgraded again, which got further confirmed when I checked for an update through the menu.

> So now you're back to running the downgraded version? Something _before_ 8.4.8? If so, while that's running, can you post the _Debug Information_ from that version?

No. I updated it... again, so it should now be the latest which according to the website is 8.4.9, tho the update still doesn't seem to have been "triggered" yet, so it's 8.4.8 right now.

Checking the "Info..." option and also running debug is making me go angry right now, because Notepad++ is yet again displayed as a 32-Bit installation, even tho I used the 64-Bit installer:
![grafik](https://user-images.githubusercontent.com/11576465/220769467-25609faa-fa10-4dc6-a6a4-360bc99dd442.png)

```
Notepad++ v8.4.8   (32-bit)
Build time : Dec 24 2022 - 19:30:12
Path : D:\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 21H2
OS Build : 19044.2604
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
```
This means it AGAIN installed a 32-Bit version for no apparent reason. And I can't figure out why this happens here... It's frustrating to say the least...

My biggest guess (Which is kinda unlikely as that has worked perfectly fine in the past 3+ years) is that my Anti-Virus and its Program-update feature (Which checks for updates to programs and if those are valid) is somehow installing the wrong updates here with the wrong Bit-version and/or final version.
Tho, it would then report said update to me...

My other guess is, that Notepad++ checks the wrong location for its version, assumes I have an outdated one, installs the new one, but in the right/wrong location(?) and when I start it, uses that outdated one... And when I manually change it, does it update and load the one in the right directory... I hope this makes sense.

Either way, I don't have any real options to check for the cause of this issue here and hope someone can help out here, as I'm so close to dropping this tool and switching to something else...

**Comment by Andre601** [comment]
Another thing I find rather puzzling: After de-installing the 32-Bit version is "Notepad++" no longer giving any results in the Start menu of Windows, even tho the list of programs lists the 64-Bit version.

Apparently was 64-Bit installed in the same directory as 32-Bit and got deinstalled too... How can a programm exist twice in the same directory?!

**Comment by alankilborn** [comment]
>it should now be the latest which according to the website is 8.4.9, tho the update still doesn't seem to have been "triggered" yet, so it's 8.4.8 right now.

Just a side note:  8.4.9 may NEVER be triggered, if the author judges that it is not solid enough for the masses to get notified.  So I would definitely not let 8.4.9 make you crazy that you aren't getting that; for practical purposes, assume 8.4.8 is the latest.

**Comment by alankilborn** [comment]
I'm going to admit I'm out of my depth in continuing to try to help with this.

Some suggestions:

* Post on the [Community](https://community.notepad-plus-plus.org/) support site; a lot more people will see your posting than will see it here, and perhaps someone can help; you can certainly describe the problem briefly and then link here (or copy chunks of text from here over to a posting there)

* If you get someone knowledgeable helping, do what they say, and only what they say, and report back results to them.  People are for some reason very tempted to go "off script" in a situation such as yours, muddying the waters for those trying to help, so that nobody can get to the root of a problem.

* Don't say things like "I'm so close to dropping this tool and switching to something else" as this just implies that Windows' notepad.exe is just as good as Notepad++ to you, so why should anyone put time and effort into trying to help you.

Good luck.
